src/main/java/com/mraof/minestuck/item/weapon/PropelEffect.java

```diff
@@ -0,0 +1,75 @@
+package com.mraof.minestuck.item.weapon;
+
+import com.mraof.minestuck.player.EnumAspect;
+import com.mraof.minestuck.player.Title;
+import com.mraof.minestuck.world.storage.ClientPlayerData;
+import com.mraof.minestuck.world.storage.PlayerSavedData;
+import net.minecraft.client.entity.player.ClientPlayerEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.*;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+
+public class PropelEffect implements ItemRightClickEffect
+{
+	public static final PropelEffect BREATH_PROPEL = new PropelEffect(3, EnumAspect.BREATH);
+	
+	private final double velocity;
+	private final EnumAspect aspect;
+	
+	public PropelEffect(double velocity, EnumAspect aspect)
+	{
+		this.velocity = velocity;
+		this.aspect = aspect;
+	}
+	
+	@Override
+	public ActionResult<ItemStack> onRightClick(World world, PlayerEntity player, Hand hand)
+	{
+		ItemStack itemStack = player.getHeldItem(hand);
+		propelAction(player, itemStack, getVelocityMod(), hand);
+		return ActionResult.resultPass(itemStack);
+	}
+	
+	private double getVelocityMod()
+	{
+		return velocity;
+	}
+	
+	void propelAction(PlayerEntity player, ItemStack stack, double velocity, Hand hand)
+	{
+		Title title = null;
+		if(player instanceof ClientPlayerEntity)
```

Comment:
Since the `ClientPlayerEntity` class doesn't exist on a dedicated server, you might want to replace this with `player.world.isRemote`

Reply:
Switched out and still works